### DISCUSSION TITLE
Audible as a Provider

### DISCUSSION THREAD

**Discussion by DellanX** [music-assistant/support#862] [Feature requests and ideas]
I Realize, that Audiobooks might not be classified as "Music"
However, they are consumed in a similar way.

There is a python package for Audible's not public API: https://github.com/mkb79/Audible
It does look like it supports URL streams and downloads, so I imagine it's integration is possible. It'd just take some knowledge I do not yet posses.

**Comment by DutchJaFO** [comment]
oh yes please 👍 
I know I can already use my Sonos system to playback audible books, but an integration into Music Assistant would be very useful.

@DellanX it usually isn't a good idea to try and support an API that isn't desinged to be public.
Companies tend to make life hard when you do so and (I think) it is against HA policy to use them for those reasons as well.

**Comment by mac-city** [comment]
I would also be very happy if audible were implemented.

**Comment by OzGav** [comment]
Audible will be available in 2.4.0b12+

**Reply by ATLAS-IND-Software** [reply]
<3

**Comment by ciB89** [comment]
Is this working? When pressing on "(re-)authenticate" nothing happens for me. Is this a me problem?

**Reply by OzGav** [reply]
No it is being looked into

**Reply by TheWhiteGandalf** [reply]
I have the same problem

**Reply by ztripez** [reply]
This should work in the latest beta.

**Reply by TheWhiteGandalf** [reply]
It works. Thank you for the great work 😃

**Comment by stevegroom** [comment]
Just installed Audible provider. The authentication is a little clunky but does work. 
I have an audible.co.uk account with 471 titles. On loading and after synchronising it only lists 50 titles.

**Reply by ztripez** [reply]
sorry for the long wait but there is a PR up to fix this issue.

**Comment by stevegroom** [comment]
How to report bugs with a specific Music Assistant provider?

**Comment by selbaer** [comment]
Hi! 
it is possible to implement authentication via username? 
Old audible account must login with username instead of amazon account. If you go to audible.com and click [sign in] you are forwarded to a login which as the option "Login with username", which is missing from the Music Assistant provider link.

![image](https://github.com/user-attachments/assets/19feac13-afe7-4b15-8ac7-ac1bc0f3d065)

This finally leads to an audible.com login screen instead of amazon. I already looked at the url's with all it's openid stuff but have to admit that this is beyond my capabilities.

Thanks in advance

**Reply by OzGav** [reply]
@ztripez Is this possible?

**Reply by ztripez** [reply]
Hm, I've missed this thread, I'll take a look.

**Reply by selbaer** [reply]
Thanks! Let me know if you need additional information.

**Comment by dschido** [comment]
Hi,

i have a german audible account and the log says:
2025-04-27 11:06:12.389 INFO (MainThread) [audible.auth] signing auth flow applied to request
2025-04-27 11:06:12.744 INFO (MainThread) [music_assistant.audible] Audible: Successfully retrieved 33 audiobooks from library

But I can't find any Audiobook in MA.

Are there any hints I can follow?

Thanks in advance

**Reply by OzGav** [reply]
You don’t say what version you are using. Make sure it is 2.5.2 or 2.6.0b4

**Reply by dschido** [reply]
sorry, I use 2.5.2.

the debug Log says:

2025-04-27 12:19:11.141 DEBUG (MainThread) [music_assistant.audible] Audible: Got 33 items (total reported by API: 0)
2025-04-27 12:19:11.141 DEBUG (MainThread) [music_assistant.audible] Skipping non-audiobook item: Die Macht des Lichts (MultiPartBook)
...
2025-04-27 12:19:11.144 DEBUG (MainThread) [music_assistant.audible] Skipping non-audiobook item: Drohende Schatten (MultiPartBook)
2025-04-27 12:19:11.144 DEBUG (MainThread) [music_assistant.audible] Audible: Processed 0 valid audiobooks on page 1
2025-04-27 12:19:11.145 DEBUG (MainThread) [music_assistant.audible] Audible: Moving to page 2 (processed: 33, total reported: 0)
2025-04-27 12:19:11.145 DEBUG (MainThread) [music_assistant.audible] Audible: Fewer than page size returned, ending pagination (processed 33 items)
2025-04-27 12:19:11.146 INFO (MainThread) [music_assistant.audible] Audible: Successfully retrieved 33 audiobooks from library


so maybe the books are not compatible to the expected format?

**Reply by OzGav** [reply]
Support for multipart books has been added to 2.6.0b4. If you want you can temporarily try that otherwise wait for the fix to come to the stable branch.

**Reply by dschido** [reply]
OK. then I will update.
Thank you very much

**Comment by tantalic** [comment]
I can't find a way to add multiple Audible providers with separate logins. Is this functionality missing?

**Reply by OzGav** [reply]
Looking at the code it seems that hasn’t been implemented

**Reply by ztripez** [reply]
I'll look into it

**Reply by ztripez** [reply]
> I can't find a way to add multiple Audible providers with separate logins. Is this functionality missing?

https://github.com/music-assistant/server/pull/2879

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
